x-pack/plugins/alerting/server/lib/get_execution_log_aggregation.test.ts

```diff
@@ -0,0 +1,842 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  getNumExecutions,
+  getExecutionLogAggregation,
+  formatExecutionLogResult,
+} from './get_execution_log_aggregation';
+
+describe('getNumExecutions', () => {
+  test('should calculate the expected number of executions in a given date range with a given schedule interval', () => {
+    expect(
+      getNumExecutions(
+        new Date('2020-12-01T00:00:00.000Z'),
+        new Date('2020-12-02T00:00:00.000Z'),
+        '1h'
+      )
+    ).toEqual(24);
+  });
+
+  test('should return 0 if dateEnd is less that dateStart', () => {
+    expect(
+      getNumExecutions(
+        new Date('2020-12-02T00:00:00.000Z'),
+        new Date('2020-12-01T00:00:00.000Z'),
+        '1h'
+      )
+    ).toEqual(0);
+  });
+
+  test('should cap numExecutions at default max buckets limit', () => {
+    expect(
+      getNumExecutions(
+        new Date('2020-12-01T00:00:00.000Z'),
+        new Date('2020-12-02T00:00:00.000Z'),
+        '1s'
+      )
+    ).toEqual(1000);
+  });
+});
+
+describe('getExecutionLogAggregation', () => {
+  test('should throw error when given bad sort field', () => {
+    expect(() => {
+      getExecutionLogAggregation({
+        numExecutions: 5,
+        page: 1,
+        perPage: 10,
+        sortField: 'notsortable',
+        sortOrder: 'asc',
+      });
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"Invalid sort field \\"notsortable\\" - must be one of [timestamp,duration]"`
```

Comment:
We will add `total_search_duration_ms` and  `es_search_duration_ms` as sorting fields in another PR.

Reply:
Yea, I thought we were just starting with timestamp and duration for sort?

Reply:
Easy enough to add these: updated in this commit: https://github.com/elastic/kibana/pull/127339/commits/dc24753802302da3523cf913620cbce72809d2e6